exporter/awsemfexporter/util.go

```diff
@@ -55,3 +59,124 @@ func replace(s, pattern string, value pdata.AttributeValue, logger *zap.Logger)
 	}
 	return strings.Replace(s, pattern, value.StringVal(), -1)
 }
+
+// getNamespace retrieves namespace for given set of metrics from user config.
+func getNamespace(rm *pdata.ResourceMetrics, namespace string) string {
+	if len(namespace) == 0 {
+		serviceName, svcNameOk := rm.Resource().Attributes().Get(conventions.AttributeServiceName)
+		serviceNamespace, svcNsOk := rm.Resource().Attributes().Get(conventions.AttributeServiceNamespace)
+		if svcNameOk && svcNsOk && serviceName.Type() == pdata.AttributeValueSTRING && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = fmt.Sprintf("%s/%s", serviceNamespace.StringVal(), serviceName.StringVal())
+		} else if svcNameOk && serviceName.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceName.StringVal()
+		} else if svcNsOk && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceNamespace.StringVal()
+		}
+	}
+
+	if len(namespace) == 0 {
+		namespace = defaultNamespace
+	}
+	return namespace
+}
+
+// getLogInfo retrieves the log group and log stream names from a given set of metrics.
+func getLogInfo(rm *pdata.ResourceMetrics, cWNamespace string, config *Config) (logGroup, logStream string) {
+	if cWNamespace != "" {
+		logGroup = fmt.Sprintf("/metrics/%s", cWNamespace)
+	}
+
+	// Override log group/stream if specified in config. However, in this case, customer won't have correlation experience
+	if len(config.LogGroupName) > 0 {
+		logGroup = replacePatterns(config.LogGroupName, rm.Resource().Attributes(), config.logger)
+	}
+	if len(config.LogStreamName) > 0 {
+		logStream = replacePatterns(config.LogStreamName, rm.Resource().Attributes(), config.logger)
+	}
+
+	return
+}
+
+// createMetricKey generates a hashed key from metric labels and additional parameters
+func createMetricKey(labels map[string]string, parameters map[string]string) string {
```

Comment:
We should use `label.Distinct` as we did in the statsd receiver.

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1670#discussion_r529226388